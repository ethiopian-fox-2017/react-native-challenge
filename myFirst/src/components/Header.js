import React, { Component } from 'react';
import { Header, Title, Right, Body } from 'native-base';


class Headernih extends Component {

  render() {
    return (
      <Header>
          <Body>
              <Title>Daily Inspiration</Title>
          </Body>
          <Right>
            <Title onPress={() => {this.props.navigation.navigate('About')} }>
              About
            </Title>
          </Right>
      </Header>
    );
  }
}

export default Headernih;
