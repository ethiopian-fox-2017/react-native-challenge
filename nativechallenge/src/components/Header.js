import React from 'react'
import {
  View,
  Text,
  StyleSheet,
  Button
} from 'react-native';

const Header = ({ name }) => {

  return (
    <View style = {styles.navbar}>
      <Button
        onPress={() => {}}
        style={styles.textBar}
        title="Back"
        color="#fff"
      />
      <Text style={styles.textBarTengah}>
        SpotifyApp
      </Text>
      <Button
        style={styles.textBar}
        title="Next"
        color="#fff"
      />

    </View>
  )
}

const styles = StyleSheet.create({
  navbar: {
    flexDirection: 'row',
    width: '100%',
    height: '10%',
    backgroundColor: '#2196F3',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  textBar: {
    fontSize: 15,
    color: '#fff',
    fontWeight: 'bold',
    padding: 10
  },
  textBarTengah: {
    fontSize: 22,
    color: '#fff',
    alignItems: 'center'
  }
});



export default Header;
