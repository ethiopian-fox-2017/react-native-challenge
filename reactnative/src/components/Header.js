import React from 'react'
import {
  View,
  Text,
  StyleSheet
} from 'react-native';

const Header = ({ name }) => {

  return (
    <View style = {styles.navbar}>
      <Text style={{textAlign: 'center', marginTop: 10, color: '#fff', fontSize: 22}}>
        { name }
      </Text>
    </View>
  )
}

const styles = StyleSheet.create({
  navbar: {
    width: '100%',
    height: '10%',
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#2196F3'
  }
});



export default Header;
