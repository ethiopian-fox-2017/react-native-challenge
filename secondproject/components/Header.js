import React from 'react'

import {
  Text,
  View
} from 'react-native'

const Header = () => {
  return (
    <View style={styles.container}>
      <Text style={styles.header}>
        Welcome to React Native!
      </Text>
    </View>
  )
}

const styles = {
  container: {
    backgroundColor: '#03A9F4'
  },
  header: {
    color: 'white',
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
}


// http://api.openweathermap.org/data/2.5/forecast?id=1642911&APPID=cd94b7c5d853fa3ef851259ac1d3347b
// {
//   "id": 1642911,
//   "name": "Jakarta",
//   "country": "ID",
//   "coord": {
//     "lon": 106.845131,
//     "lat": -6.21462
//   }
// },

export default Header