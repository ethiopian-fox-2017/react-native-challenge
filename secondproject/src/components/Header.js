import React from 'react'

import {
  Text,
  View
} from 'react-native'

const Header = () => {
  return (
    <View style={styles.container}>
      <Text style={styles.header}>
        Welcome to React Native!
      </Text>
    </View>
  )
}

const styles = {
  container: {
    backgroundColor: '#03A9F4'
  },
  header: {
    color: 'white',
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
}

export default Header