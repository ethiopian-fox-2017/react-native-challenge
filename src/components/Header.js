import React from 'react'
import {StatusBar, View, Text, StyleSheet} from 'react-native'
import ToolbarAndroid from 'ToolbarAndroid'


var toolbarActions = [
  {title: 'Tambah Data'},
  {title: 'List Data'},
];

var styles = StyleSheet.create({
  toolbar: {
    backgroundColor: '#FFB300',
    height: 56,
  }
})

class Header extends React.Component{
  constructor(props){
    super(props)
    this.state = {
      actionText : '',
      toolbarSwitch : '',
      colorProps : {
        titleColor : '',
        subtitleColor : '',
      }
    }
  }
  getInitialState() {
    return {
      actionText: 'Example app with toolbar component',
      toolbarSwitch: false,
      colorProps: {
      titleColor: '#3b5998',
      subtitleColor: '#6a7180',
      },
    };
  }

  render(){
    return (
      <View>
      <ToolbarAndroid
        actions={toolbarActions}
        onActionSelected={this._onActionSelected}
        onIconClicked={() => this.setState({actionText: 'Icon clicked'})}
        style={styles.toolbar}
        title="Cari Makan" />
      </View>
    )
  }

  _onActionSelected(position) {
    console.log('ada',position);
  }
}

export default Header
