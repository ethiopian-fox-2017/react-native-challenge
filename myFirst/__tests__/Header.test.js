import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import Header from '../src/components/Header';
// Note: test renderer must be required after react-native.


test('renders correctly', () => {
  const tree = shallow(
      <Header />
  );

  expect(tree).toMatchSnapshot();
});
