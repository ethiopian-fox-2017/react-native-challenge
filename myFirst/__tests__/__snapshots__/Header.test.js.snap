// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(Header)
    virtual={undefined}
>
    <Styled(Body)
        virtual={undefined}
    >
        <Styled(Title)
            virtual={undefined}
        >
            Daily Inspiration
        </Styled(Title)>
    </Styled(Body)>
    <Styled(Right)
        virtual={undefined}
    >
        <Styled(Title)
            onPress={[Function]}
            virtual={undefined}
        >
            About
        </Styled(Title)>
    </Styled(Right)>
</Styled(Header)>,
  "nodes": Array [
    <Styled(Header)
      virtual={undefined}
>
      <Styled(Body)
            virtual={undefined}
      >
            <Styled(Title)
                  virtual={undefined}
            >
                  Daily Inspiration
            </Styled(Title)>
      </Styled(Body)>
      <Styled(Right)
            virtual={undefined}
      >
            <Styled(Title)
                  onPress={[Function]}
                  virtual={undefined}
            >
                  About
            </Styled(Title)>
      </Styled(Right)>
</Styled(Header)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Headernih />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Headernih {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(Header)
          virtual={undefined}
>
          <Styled(Body)
                    virtual={undefined}
          >
                    <Styled(Title)
                              virtual={undefined}
                    >
                              Daily Inspiration
                    </Styled(Title)>
          </Styled(Body)>
          <Styled(Right)
                    virtual={undefined}
          >
                    <Styled(Title)
                              onPress={[Function]}
                              virtual={undefined}
                    >
                              About
                    </Styled(Title)>
          </Styled(Right)>
</Styled(Header)>,
        "_debugID": 2,
        "_renderedOutput": <Styled(Header)
          virtual={undefined}
>
          <Styled(Body)
                    virtual={undefined}
          >
                    <Styled(Title)
                              virtual={undefined}
                    >
                              Daily Inspiration
                    </Styled(Title)>
          </Styled(Body)>
          <Styled(Right)
                    virtual={undefined}
          >
                    <Styled(Title)
                              onPress={[Function]}
                              virtual={undefined}
                    >
                              About
                    </Styled(Title)>
          </Styled(Right)>
</Styled(Header)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Headernih />,
}
`;
